### PR TITLE
I tried to make an abstract fabric method.

### PR DESCRIPTION
**Pull request by Y-Pazniak**
I have three groups of families objects.
Depending on type of object in family (safe or not) application generates group of animals for pet store or for zoo.

### CODE REVIEW

File: src/main/java/org/example/Main.java [hunk 1]
```diff
@@ -1,30 +1,25 @@
...
-import org.example.cage.GiraffesCage;
-import org.example.cage.WolfsCage;
-import org.example.cage.ZooCage;
+import org.example.spaces.CreaturesFactory;
+import org.example.spaces.PetShopFactory;
+import org.example.spaces.ZooFactory;
 
 public class Main {
-    private static ZooCage zooCage;
+    private static final boolean needAnimalsForPetShop = false;
```
Comment (MaximPilipeiko) [inline src/main/java/org/example/Main.java:8]: You could take it from main args to avoid hardcode

File: src/main/java/org/example/Main.java [hunk 2]
```diff
@@ -1,30 +1,25 @@
...
     }
 
-    private static void doFeedersWork() {
-        System.out.println(zooCage.askForFeeder().feed());
-        System.out.println(zooCage.askForFeeder().clean());
+    private static Application configureApplication() {
+        CreaturesFactory creaturesFactory;
 
-    }
+        if (needAnimalsForPetShop) {
```
Comment (MaximPilipeiko) [inline src/main/java/org/example/Main.java:19]: You can put all logic of this method in a single class
```suggestion
       return needAnimalsForPetShop ? new PetShopFactory() :  new ZooFactory()
```

File: src/main/java/org/example/animals/Animals.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.example.animals;
+
+public interface Animals {
```
Comment (MaximPilipeiko) [inline src/main/java/org/example/animals/Animals.java:3]:
```suggestion
public interface Animal {
```

File: src/main/java/org/example/animals/DomesticCat.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package org.example.animals;
+
+public class DomesticCat implements Animals{
```
Comment (MaximPilipeiko) [inline src/main/java/org/example/animals/DomesticCat.java:3]: Please, don't forget to format code before committing.
```suggestion
public class DomesticCat implements Animals {
```

File: src/main/java/org/example/birds/Birds.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package org.example.birds;
+
+public interface Birds {
```
Comment (MaximPilipeiko) [inline src/main/java/org/example/birds/Birds.java:3]:
```suggestion
public interface Bird {
```

**Review by MaximPilipeiko — APPROVED** [review]

File: src/main/java/org/example/Main.java (hunk 1 above)
Comment (Y-Pazniak) [reply src/main/java/org/example/Main.java]: Fixed.

File: src/main/java/org/example/Main.java (hunk 2 above)
Comment (Y-Pazniak) [reply src/main/java/org/example/Main.java]: Done.

File: src/main/java/org/example/animals/Animals.java (hunk 3 above)
Comment (Y-Pazniak) [reply src/main/java/org/example/animals/Animals.java]: Fixed.

File: src/main/java/org/example/animals/DomesticCat.java (hunk 4 above)
Comment (Y-Pazniak) [reply src/main/java/org/example/animals/DomesticCat.java]: Sorry. Have done it recently

File: src/main/java/org/example/birds/Birds.java (hunk 5 above)
Comment (Y-Pazniak) [reply src/main/java/org/example/birds/Birds.java]: Fixed

**Pull request merged**